lightning/src/ln/reorg_tests.rs

```diff
@@ -196,16 +196,28 @@ fn do_test_unconf_chan(reload_node: bool, reorg_after_reload: bool, connect_styl
 	let mut nodes = create_network(2, &node_cfgs, &node_chanmgrs);
 	*nodes[0].connect_style.borrow_mut() = connect_style;
 
-	let chan_id = create_announced_chan_between_nodes(&nodes, 0, 1, InitFeatures::known(), InitFeatures::known()).2;
+	let chan = create_announced_chan_between_nodes(&nodes, 0, 1, InitFeatures::known(), InitFeatures::known());
 
 	let channel_state = nodes[0].node.channel_state.lock().unwrap();
 	assert_eq!(channel_state.by_id.len(), 1);
 	assert_eq!(channel_state.short_to_id.len(), 1);
 	mem::drop(channel_state);
 
 	if !reorg_after_reload {
-		disconnect_all_blocks(&nodes[0]);
-		check_closed_broadcast!(nodes[0], true);
+		if use_funding_unconfirmed {
+			let relevant_txids = nodes[0].node.get_relevant_txids();
+			assert_eq!(relevant_txids.len(), 1);
+			assert_eq!(&relevant_txids[..], &[chan.3.txid()]);
+			nodes[0].node.transaction_unconfirmed(&relevant_txids[0]);
+		} else {
+			disconnect_all_blocks(&nodes[0]);
+		}
+		if connect_style == ConnectStyle::FullBlockViaListen && !use_funding_unconfirmed {
+			handle_announce_close_broadcast_events(&nodes, 0, 1, true, "Funding transaction was un-confirmed. Locked at 6 confs, now have 2 confs.");
+		} else {
+			handle_announce_close_broadcast_events(&nodes, 0, 1, true, "Funding transaction was un-confirmed. Locked at 6 confs, now have 0 confs.");
+		}
+		check_added_monitors!(nodes[1], 1);
```

Comment:
This test keeps growing... Would it make sense to separate the various behaviors into their own tests at this point? It isn't clear to me that the cartesian product needs to be tested.

Reply:
Hmm, we could clone the test and run the full set of connection styles against something that drops the reload code, but we'd still end up with basically the same thing duplicated. I definitely think we need to test the set of full connection/disconnection styles, since I found unexpected bugs in this test when doing so across the full test suite :).